### DISCUSSION TITLE
Read Open-ADAS files for atomic data?

### DISCUSSION THREAD

**Discussion by drsmith48** [PlasmaPy/PlasmaPy#2404] [Feature Ideas & Requests]
Hi PlasmaPy community.  Would reading Open-ADAS atomic data files be an appropriate feature for PlasmaPy? See here, for example: https://open.adas.ac.uk/adf04. Thanks.

**Comment by drsmith48** [comment]
@namurphy @StanczakDominik 
Similarly, would the functionality of ColRadPy (https://github.com/johnson-c/ColRadPy) be appropriate for inclusion in PlasmaPy?
Generalized collisional radiative modeling:
https://doi.org/10.1088/0741-3335/48/2/007
https://doi.org/10.1016/j.nme.2019.01.013

**Comment by namurphy** [comment]
Thank you for bringing this up! I think all of these capabilities would are really important and should either be in PlasmaPy or at least interoperable with it. At the very least, it would be great to make sure that there is interoperability. Probably the biggest limitation would be for the people who work on these packages/databases to find a spare month or three to twenty work on this.

A complicating factor is [licensing compatibility](https://en.wikipedia.org/wiki/License_compatibility). If I remember correctly, the [terms and conditions of OPEN-ADAS](https://open.adas.ac.uk/terms-and-conditions) prohibit commercial use without prior authorization. (The link seems to be down at the moment so I can't check.) Their license is incompatible with PlasmaPy's BSD 3-clause license and also the [Open Source Definition](https://opensource.org/osd/).  Similarly, ColRadPy is licensed under the copyleft GPLv3 license, so it would need to be relicensed to become part of PlasmaPy. It wouldn't be a problem for ColRadPy to depend on PlasmaPy or vice versa, though.

Cross-reference: #52.

**Reply by namurphy** [reply]
There are also projects like [Cherab](https://github.com/cherab) too, which may fill some of this need. It seems worthwhile to start a conversation with them about what it would take to make things interoperable with each other.  Thank you again for bringing this up!
